### ISSUE TITLE
PDF generation for powerpoint is failing

### ISSUE THREAD

**Issue by ishanig11** [OfficeDev/office-js#5006]
Hi,

PDF generation for powerpoint using office.js was working fine in last month.
This month this functionality is failing.

 Office.context.document.getFileAsync(
        fileType,
        fileType !== Office.FileType.Pdf ? { sliceSize: 65536 } : null,
        async function (result) {
          if (result.status === Office.AsyncResultStatus.Succeeded) {
            const myFile = result.value;

![image](https://github.com/user-attachments/assets/6232ef6d-a89f-43be-843c-7be2d9e0382a)

please check this priority as this is blocker issue.

**Comment by RuizhiSunMS** [comment]
ACK. I'm reaching out to PPT team and will let them take a look. Thx for your submit.

**Comment by EsterBergen** [comment]
@ishanig11 - Thanks for reporting it. I've filed a bug (9484402) and will let you know here once it's updated. Thanks!

**Comment by EsterBergen** [comment]
@ishanig11 - Can you confirm if you're closing the file after use? Are you calling file.closeAsync()?

Here's an evaluation from one of our engineers: "Using the same code snippet they provided and leaving out closing the file after, I can only repro the issue after multiple running it twice. I believe the issue is they may be opening file descriptors, but since they do not close them after using them they are locked open. Any future calls after would then fail due to none being available.

Here is a link to an example of closing files after use: https://learn.microsoft.com/en-us/javascript/api/office/office.document?view=common-js-preview&redirectedfrom=MSDN#:~:text=myFile.closeAsync()%3b 

Here is another report of the problem and the comment states the same thing (it was the issue): https://stackoverflow.com/questions/25931764/office-context-document-getfileasync-throwing-errors
"

Let us know if the issue still exists once you call file.closeAsync()

**Comment by ishanig11** [comment]
Yes, we are closing file after use. Same code is working fine for Word, Excel. But failing for powerpoint.

**Comment by ihostbert** [comment]
Any updates on this, we are seeing the same issue

**Comment by ExecutiveCoder** [comment]
Same issue - please update

**Comment by ivanterekh** [comment]
I am experiencing the same issue. I have a very simple function:
```
function savePDF() {
  const format = Office.FileType.Pdf;
  Office.context.document.getFileAsync(format, { sliceSize: 1024 * 32 }, function (result) {
    console.log("result", result);
    if (result.status === Office.AsyncResultStatus.Succeeded) {
      let file: Office.File = result.value;
      setMessage("success");
      getSliceAsync(file, 0, file.sliceCount, true, [], 0, sendFileSlices);
    } else {
      setMessage(
        "failed to create PDF:\n" + result.error.code + "  " + result.error.name + "\n" + result.error.message
      );
    }
  });
}
```
The logged result error:
```
{
    "name": "Intern feil",
    "message": "Det har oppstått en intern feil",
    "code": 5001
}
```

Steps to reproduce:
1. Open a new window in incognito mode in chrome
2. Log in to Office 365
3. Create empty presentation and edit text on default slide
4. Add an Add-in with "save to PDF" functionality
5. Trigger the function `saveAsPDF` mentioned above
6. Receive error 5001

This functionality works fine for `Office.FileType.Compressed`. However, it fails for `Office.FileType.Pdf`.

It was mentioned before that not closing the file might be a problem. It is definitely not the case since issue persists in a freshly created presentation in a freshly opened incognito window. I am not even able to get to a place where I have a file to close.

Platform: PowerPoint Web
Browser: Chrome
OS: Ubuntu and Windows
Build: 16.0.18522.40506

**Comment by sebacardello** [comment]
👋 Recently discovered we're facing the same issue: the same code stop failing once changing `Office.FileType.Pdf` to `Office.FileType.Compressed`. It isn't related to open file descriptors. BTW, also tried the preview (beta) version with no luck.

Any update on this?

**Comment by skribble-freddy** [comment]
Hi any updates on this, we are seeing the same issue

**Comment by skribble-freddy** [comment]
This seems to be working now. Could someone please confirm?

**Comment by directable** [comment]
this appears to be broken again with the online version of powerpoint . it works fine in the desktop version.  It is returning "internal server error"
